service/kernel/env_detect.cpp

```diff
@@ -144,15 +144,11 @@ DAAL_EXPORT void daal::services::Environment::initNumberOfThreads()
         const size_t ncores = daal::internal::Service<>::serv_get_ncpus() * daal::internal::Service<>::serv_get_ncorespercpu();
 
         /*  Re-set number of threads if ncores is valid */
-        if ((ncores > 0) && (ncores <= _daal_threader_get_max_threads()))
+        if ((ncores > 0) && (ncores < _daal_threader_get_max_threads()))
         {
             daal::services::Environment::setNumberOfThreads(ncores);
         }
     }
-    else
-    {
-        daal::services::Environment::setNumberOfThreads(_daal_threader_get_max_threads());
-    }
     isInit = true;
```

Comment:
Does it solve the original problem?

Reply:
Actually, solution is the using global_control instead of task_scheduler_init for setting number of threads. This diff between commits is just shows that previous changes were reverted =)